hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllBackupOperation.java

```diff
@@ -75,10 +78,26 @@ public void run() throws Exception {
             for (Map.Entry<Data, CacheRecord> entry : cacheRecords.entrySet()) {
                 CacheRecord record = entry.getValue();
                 cache.putRecord(entry.getKey(), record);
+
+                publishWanEvent(entry.getKey(), record);
             }
         }
     }
 
+    private void publishWanEvent(Data key, CacheRecord record) {
+        if (cache.isWanReplicationEnabled()) {
+            ICacheService service = getService();
+            final CacheWanEventPublisher publisher = service.getCacheWanEventPublisher();
+            final CacheEntryView<Data, Data> view = CacheEntryViews.createDefaultEntryView(
+                    key, toData(record.getValue()), record);
+            publisher.publishWanReplicationUpdate(name, view);
+        }
+    }
+
+    private Data toData(Object o) {
+        return getNodeEngine().getSerializationService().toData(o);
```

Comment:
was the intent to create heap-data from native one? if yes, this shouldn't work, instead `ToHeapDataConverter` can be used.